### ISSUE TITLE
Error on output

### ISSUE THREAD

**Issue by micheleietri** [daryllabar/DLaB.Xrm.XrmToolBoxTools#84]
When running the creation, i always got this as output on console.

CRM Authentication Type Detected: ActiveDirectory
Executing 
"C:\Users\michele.ietri\AppData\Roaming\MscrmTools\XrmToolBox\Plugins\CrmSvcUtil Ref\crmsvcutil.exe"^
/namespace:"CrmEarlyBound"^
/out:"C:\Users\michele.ietri\AppData\Roaming\MscrmTools\XrmToolBox\Settings\CrmSvcUtil Ref\Entities\XrmServiceContext.cs"^
/servicecontextname:"XrmServiceContext"^
/codecustomization:"DLaB.CrmSvcUtilExtensions.Entity.CustomizeCodeDomService,DLaB.CrmSvcUtilExtensions"^
/codegenerationservice:"DLaB.CrmSvcUtilExtensions.Entity.CustomCodeGenerationService,DLaB.CrmSvcUtilExtensions"^
/codewriterfilter:"DLaB.CrmSvcUtilExtensions.Entity.CodeWriterFilterService,DLaB.CrmSvcUtilExtensions"^
/namingservice:"DLaB.CrmSvcUtilExtensions.NamingService,DLaB.CrmSvcUtilExtensions"^
/metadataproviderservice:"DLaB.CrmSvcUtilExtensions.Entity.MetadataProviderService,DLaB.CrmSvcUtilExtensions"^
/connectionstring:"Url=http://mycorrecturl;Domain=domanin;UserName=username;Password=password" CrmSvcUtil : CRM Service Utility [Version 8.2.0.749]
¸ 2015 Microsoft Corporation.  All rights reserved.

Options:
 /nologo
  Suppresses the banner.

 /language:<language>
  The language to use for the generated proxy code.  This can be either 'CS' or 'VB'.  The default language is 'CS'.  Short form is '/l:'.

 /url:<url>
  A url or path to the SDK endpoint to contact for metadata.

 /out:<filename>
  The filename for the generated proxy code.  Short form is '/o:'.

 /namespace:<namespace>
  The namespace for the generated proxy code.  The default namespace is the global namespace.  Short form is '/n:'.

 /username:<username>
  Username to use when connecting to the server for authentication.  Short form is '/u:'.

 /password:<password>
  Password to use when connecting to the server for authentication.  Short form is '/p:'.

 /domain:<domain>
  Domain to authenticate against when connecting to the server.  Short form is '/d:'.

 /serviceContextName:<service context name>
  The name for the generated service context. If a value is passed in, it will be used for the Service Context.  If not, no Service Context will be generated

 /help
  Show this usage message.  Short form is '/?'.

 /generateActions
  Generate wrapper classes for custom actions  Short form is '/a'.

 /deviceid:<deviceid>
  Device ID to use when connecting to the online server for authentication.  Short form is '/di:'.

 /devicepassword:<devicepassword>
  Device Password to use when connecting to the online server for authentication.  Short form is '/dp:'.

Example:
crmsvcutil.exe /url:http://localhost/Organization1/XRMServices/2011/Organization.svc /out:GeneratedCode.cs

Output file was not updated or not found!  C:\Users\michele.ietri\AppData\Roaming\MscrmTools\XrmToolBox\Settings\CrmSvcUtil Ref\Entities\XrmServiceContext.cs

the strange thing is that it want to create a file XrmServiceContext.cs when i told it to create ine file per entity

**Comment by daryllabar** [comment]
hmm... looks like maybe I need to add the AuthType to the Connection String?

Can you try this in a command prompt?:
```
"C:\Users\michele.ietri\AppData\Roaming\MscrmTools\XrmToolBox\Plugins\CrmSvcUtil Ref\crmsvcutil.exe" /namespace:"CrmEarlyBound" /out:"C:\Users\michele.ietri\AppData\Roaming\MscrmTools\XrmToolBox\Settings\CrmSvcUtil Ref\Entities\XrmServiceContext.cs" /servicecontextname:"XrmServiceContext" /codecustomization:"DLaB.CrmSvcUtilExtensions.Entity.CustomizeCodeDomService,DLaB.CrmSvcUtilExtensions" /codegenerationservice:"DLaB.CrmSvcUtilExtensions.Entity.CustomCodeGenerationService,DLaB.CrmSvcUtilExtensions" /codewriterfilter:"DLaB.CrmSvcUtilExtensions.Entity.CodeWriterFilterService,DLaB.CrmSvcUtilExtensions" /namingservice:"DLaB.CrmSvcUtilExtensions.NamingService,DLaB.CrmSvcUtilExtensions" /metadataproviderservice:"DLaB.CrmSvcUtilExtensions.Entity.MetadataProviderService,DLaB.CrmSvcUtilExtensions" /connectionstring:"AuthType=AD;Url=http://mycorrecturl;Domain=domanin;UserName=username;Password=password" 
```

**Comment by micheleietri** [comment]
same issue

**Comment by daryllabar** [comment]
How about?

```
"C:\Users\michele.ietri\AppData\Roaming\MscrmTools\XrmToolBox\Plugins\CrmSvcUtil Ref\crmsvcutil.exe" /namespace:"CrmEarlyBound" /out:"C:\Users\michele.ietri\AppData\Roaming\MscrmTools\XrmToolBox\Settings\CrmSvcUtil Ref\Entities\XrmServiceContext.cs" /servicecontextname:"XrmServiceContext" /connectionstring:"AuthType=AD;Url=http://mycorrecturl;Domain=domain;UserName=username;Password=password" 
```

**Comment by micheleietri** [comment]
still the same

**Comment by anubhavbajpai** [comment]
Hi team,

There seems some issue with the connection string tag that is being used.

I ran the same connectionstring (users please note that the connection string url is actually missing the organization.svc path) on command prompt which is giving error.

Then I tried with the correct options like /url:<my organization service url> /username:<domain\username> /password:<password>and I kept all the other parameters same which were being passed in the tool.

I was able to successfully extract the class files.

I hope this info will help the developers as well as other users while this issue is being fixed.

Thanks & Regards,
Anubhav

**Comment by micheleietri** [comment]
i tried your same way, but i got the same output!

**Comment by daryllabar** [comment]
@anubhavbajpai Can you give me the output that the tool is generating, and the output that you changed it to, to be successful?  Feel free to obfuscate your org in the URL, UserName, and Password, but show the rest please.

**Comment by bcolpaert** [comment]
Hi daryllabar, 

If we change the command and replace 
`/connectionstring:"Url=http://mycorrecturl;Domain=domanin;UserName=username;Password=password" 
`with
`/url:organization_service_url /username:<domain\username> /password:_password`
the file creation works

**Comment by micheleietri** [comment]
I can confirm, using /url:organization_service_url /username:<domain\username> /password:_password instead of connection string works fine

**Comment by snugglesftw** [comment]
@daryllabar I think it's changed for 365 instance or the crmsvc util has changed. 
I need to use /url:svc /username:etc formats

**Comment by somefella123** [comment]
+1 - if you look at the latest CrmSvcUtil it seems to have dropped support for /connectionstring as a parameter.

**Comment by daryllabar** [comment]
I don't want to be creating work-a-rounds for issues that are due to bugs in the tooling.  If the URL parameters are unable to work due to the newest version of the CrmSvcUtil not working, please submit a ticket through your office 365 admin.  Microsoft should offer support on what they have documented.  As a work around, try manually replacing the CrmSvcUtil with an older version.  For some reason, it's working just fine for me an the instance I use, so I'm unable to do this myself.

**Comment by daryllabar** [comment]
Ok, just got word back from Microsoft, that the have an issue with the Nuget Package, and will be rolling out an 8.2.0.3 version that should fix this.  I'll update it when I'm notified.

**Comment by daryllabar** [comment]
Can someone give [this version](https://drive.google.com/uc?export=download&id=0B0By9PBv_3cDaWE2ZGVtbV9kWVk) a try?

**Comment by MarcusNaylor** [comment]
That works!  thanks

**Comment by daryllabar** [comment]
I've had multiple people tell me this is fixed now in 1.2017.3.3  Let me know if it's not.
